### PR TITLE
[ownership] Enable GlobalOpt on ossa and add ossa tests

### PR DESCRIPTION
**Pull request by meg-gupta**
GlobalOpt works mostly on trivial values (there are special cases for `ObjectInst` and `ValueToBridgeObjectInst`). 
`optimizeGlobalAccess` is explicitly turned off for non-trivial values. `optimizeInitializer` calls  `SILGlobalVariable::isValidStaticInitializerInst` which limits it to mostly trivial values except for special cases for `ObjectInst` and `ValueToBridgeObjectInst`.

This changes adds GlobalOpt tests for ossa and enables GlobalOpt on ossa

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

**Review by atrick — COMMENTED** [review]
I'm not sure what's being tested. Can you please add some comments to globalopt_trivial_nontrivial.sil?

**Comment by meg-gupta** [comment]
@swift-ci smoke test

File: test/SILOptimizer/globalopt_trivial_nontrivial.sil [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import SwiftShims
+
+public struct TStruct {
+  @_hasStorage let x: Int32 { get }
+  init(x: Int32)
+}
+
+@_hasStorage @_hasInitialValue let trivialglobal: TStruct { get }
+
+@_hasMissingDesignatedInitializers public class TClass {
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply test/SILOptimizer/globalopt_trivial_nontrivial.sil]: Done.

**Comment by meg-gupta** [comment]
@atrick Added comments to `globalopt_trivial_nontrivial.sil`. I added this test just to show no-op of global opt on non-trivial values.

**Review by atrick — APPROVED** [review]
Thanks!

**Pull request merged**